04_data_structures/07_roaring_bitmaps/uncompressed.go

```diff
@@ -1,32 +1,120 @@
 package bitmap
 
+import (
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
 const wordSize = 64
 
 type uncompressedBitmap struct {
 	data []uint64
 }
 
+func (bitMap *uncompressedBitmap) String() string {
+	res := ""
+	for i, bucket := range bitMap.data {
+		if bucket != 0 {
+			res += fmt.Sprintf("bucket %7d=%064b\n", i, bucket)
+		}
+	}
+	return res
+}
+
+var atom = zap.NewAtomicLevel()
+
+var logger = zap.New(zapcore.NewCore(
+	zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()),
+	zapcore.Lock(os.Stdout),
+	atom,
+))
+var sugar = logger.Sugar()
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
 func newUncompressedBitmap() *uncompressedBitmap {
-	return &uncompressedBitmap{}
+	return &uncompressedBitmap{
+		data: []uint64{},
+		// TODO abozhenko for robot-dreams:
```

Comment:
@robot-dreams

Reply:
I suppose it depends on your use case. If I was building this I would probably give the user the optional ability to add a hint (like how when you can do `make([]int, 8, 1024)` in go to set the capacity).